### ISSUE TITLE
Mac compile Undefined symbols for architecture for x86_64

### ISSUE THREAD

**Issue by xiangjianmeng** [zeromq/cppzmq#253]
Undefined symbols for architecture x86_64:
  "_zmq_msg_routing_id", referenced from:
      zmq::message_t::routing_id() const in message.cpp.o
  "_zmq_msg_set_routing_id", referenced from:
      zmq::message_t::set_routing_id(unsigned int) in message.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by laplaceyang** [comment]
What version of zmq lib are you using? The zmq::message_t::set_routing_id should be used under macro ZMQ_BUILD_DRAFT_API opened in both zmq lib and cppzmq side.

**Comment by qings2329** [comment]
@laplaceyang  i have the same problem.  My zmq version is 4.2.5  .   Which version should I install ?

**Comment by kurdybacha** [comment]
@laplaceyang how do you install libzmq and how do use `cppzmq`? Are you compiling `cppzmq` project from cloned git repo?

**Comment by fenggeren** [comment]
i have the same problem too
zeromq version 4.2.5.   use  'brew install zeromq'  installed.

**Comment by sigiesec** [comment]
Please provide some more information on how to reproduce this. How do you acquire cppzmq and what command generated the above-mentioned errors?

**Comment by cachedout** [comment]
I can confirm this. I cloned the repo and attempted a build from the HEAD. From a newly created `build` directory the command to replicate are:

```
cmake ..
sudo make -j4 install
```

**Comment by kurdybacha** [comment]
@cachedout @xiangjianmeng @laplaceyang building `cppzmq` from cloned repo enables drafts API by default. Please try:
```
cmake -DENABLE_DRAFTS=OFF ..
sudo make -j4 install
```

**Comment by dendisuhubdy** [comment]
@kurdybacha that solved my problem on Mac 10.14 too, thanks!
